### ISSUE TITLE
A lot of timeouts

### ISSUE THREAD

**Issue by vixez** [dart-tools/dio_http#4]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/dart-tools/dio_http/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   flutter   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |   all   |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |  5.0.4    |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |    Android Studio Arctic Fox 2020.3.1 |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |    50%  |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |   /   |
| Demo project link              | e.g. link to a demo project that highlights the issue |   /   |

### Issue Description and Steps

I get a lot of errors en Sentry and sometimes during debugging that there is a timeout, but this timeout seems to be false.
It seems like this happens in the [original Dio](https://github.com/flutterchina/dio/issues/1230) package as well.
This happens quite frequently, so it would be nice if this is fxied.

**Comment by jgoyvaerts** [comment]
Can you describe your usecase when this is happening? The original issue you linked talks about timeouts on requests that happen in a loop, is this also your case or does it happen on random single requests as well? Are the requests triggered by user interaction or by some other mechanism?

**Comment by vixez** [comment]
It can happen at any time:

- Requests made in the background (eg an update of translations)
- Requests made by the user (user presses a button, and the app then waits for the response)

It can be while a single request is made, or when a bunch of requests are made at the same time.

I can't seem find a pattern in it.

**Comment by jgoyvaerts** [comment]
Keep in mind that if this is on mobile in production, unstable connections are the most likely source of timeouts, and they should be handled by you in an intelligent way (retry behavior, etc). 

Is it possible that this is the behavior you're seeing?

**Comment by vixez** [comment]
We have it during debug and in production, on mobile and web.

The app indeed catches those errors and handles it gracefully, it's just not a good user experience.

**Comment by jgoyvaerts** [comment]
Let's focus on the web case. If you can provide me with a screenshot of a case where the request succeeded in chrome (you can see it in the network tab) but failed in Dio, I can investigate it further. But for now, I have to assume this is just random network unstability and not related to Dio itself

**Comment by vixez** [comment]
Alright, when it happens again on web I'll post an update here (we are focussing on the mobile apps right now, so it might take some time).
Thanks for the follow up in advance.

**Comment by markphillips100** [comment]
Are you using the http2_adapter and Dio as a singleton?  If so, see my comment about something I discovered in the adapter's connection manager [here](https://github.com/flutterchina/dio/issues/1230#issuecomment-954760521).  This repo and the original have the same relevant code so applies here too.

**Comment by thedang** [comment]
I used Dio as a singleton, but dont use http2_adapter, and sometimes it's happen, 

i had stop and start flutter again, it's gone. And  i dont know why.

I'll try your solution.

Thanks !

**Comment by unacorbatanegra** [comment]
@thedang  dit it work for u?

**Comment by thedang** [comment]
After checking, I found that if I turn off firebase crashlytics everything is ok,
Can you try solution

**Comment by jgoyvaerts** [comment]
This package is now archived. This was originally forked because the owner of the [Dio](https://github.com/flutterchina/dio) package was inactive, but he has sinced returned from inactiveness and has added some more contributors to prevent this from happening again in the future. All future development continues over on [Dio](https://github.com/flutterchina/dio).
